frontend/components/employee/employee.js

```diff
@@ -0,0 +1,201 @@
+import {Grid} from "@mui/material";
+import * as React from "react";
+import {useState} from "react";
+import {CustomDatePicker, CustomAutoComplete, InputTextField, DropDownList} from '../form/formComponent';
+import Button from "@mui/material/Button";
+import DialogActions from "@mui/material/DialogActions";
+
+
+const EmployeeForm = (props) => {
+
+    const titleList = [
+        {id: 0, value: "General Practice"},
+        {id: 1, value: "Acupuncture"},
+        {id: 2, value: "Manager"}
+    ];
+    const genderList = [
+        {id: '0', value: 'male', title: 'Male'},
+        {id: '1', value: 'female', title: 'Female'},
+        {id: '2', value: 'na', title: 'N/A'},
+    ];
+    const {
+        handleClose,
+        mode,
+        initValues,
+        tabValue,
+        editEmployee,
+        addEmployee,
+        serviceList,
+        serviceEmployeeList
+    } = props;
+    const [employeeValue, setEmployeeValue] = useState(initValues);
+    const [errorMessage, setErrorMessage] = useState([]);
+
+
+    const addNewEmployee = () => {
+        if (validate()) {
+            let service_ids = [];
+            for (let service of employeeValue.services) {
+                let idArray = service.durations_prices.map(durationprice => durationprice.id)
+                console.log(idArray);
+                service_ids = [...service_ids, ...idArray];
+            }
+            employeeValue.service_ids = service_ids;
+            addEmployee(employeeValue);
+            handleClose();
+        }
+    };
+    const saveEditEmployee = () => {
+        let service_ids = [];
+        for (let service of employeeValue.services) {
+            let idArray = service.durations_prices.map(durationprice => durationprice.id)
+            console.log(idArray);
+            service_ids = [...service_ids, ...idArray];
+        }
+        employeeValue.service_ids = service_ids;
+        editEmployee(employeeValue);
+    }
+    const handleSetEmployeeValue = (obj) => {
+        const {name, value} = obj.target;
+        setEmployeeValue({...employeeValue, [name]: value});
+    };
+
+
+    const validate = () => {
+        let temp = {}
```

Comment:
I suggest avoid using `temp` as a variable name. It doesn't say anything useful. Also note that `temp` does not get reassigned, so it's recommended to use `const` instead.

Reply:
I changed the variable name and type.